g11n-ws/modules/md-restful-api-i18n/src/main/java/com/vmware/vip/i18n/api/base/TranslationProductAction.java

```diff
@@ -91,9 +97,159 @@ public APIResponseDTO getMultTrans(String productName, String version,String pse
             translationDTO.setPseudo(new Boolean(pseudo));
         }
         translationDTO = multipleComponentsService.getMultiComponentsTranslation(translationDTO);
-        return super.handleResponse(APIResponseStatus.OK, translationDTO);
+        return translationDTO;
     }
+    
+    
+    public APIResponseDTO getMultTrans(String productName, String version, String componentsStr, String localesStr, String pseudo,
+            HttpServletRequest req)  throws Exception {
+        if(StringUtils.isEmpty(componentsStr) && StringUtils.isEmpty(localesStr)) {
+            return  super.handleResponse(APIResponseStatus.OK, getAllCompTrans( productName,  version, pseudo, req)) ;
+        }else {
+            return getV2MultipleComponentsTrans(productName, componentsStr,  version,  localesStr,  pseudo, req);
```

Comment:
should add validation to componentsStr and localesStr to make sure it's split by comma, otherwise throw exception.

Reply:
process the business as following: if don't have comma, the components  String will be treat as a component name, the locales string will be treat as a locale